Boss/Mod/RpcWrapper.cpp

```diff
@@ -0,0 +1,101 @@
+#include"Boss/Mod/Rpc.hpp"
+#include"Boss/Mod/RpcWrapper.hpp"
+#include"Boss/Msg/Init.hpp"
+#include"Boss/Msg/RequestRpcCommand.hpp"
+#include"Boss/Msg/ResponseRpcCommand.hpp"
+#include"Boss/concurrent.hpp"
+#include"Ev/foreach.hpp"
+#include"S/Bus.hpp"
+#include"Util/make_unique.hpp"
+#include<vector>
+
+namespace Boss { namespace Mod {
+
+class RpcWrapper::Impl {
+private:
+	S::Bus& bus;
+
+	/*~ The core object that will actually handle
+	 * the command.
+	 */
+	Boss::Mod::Rpc* rpc;
```

Comment:
Why not a smart pointer?

Reply:
`Boss::Mod::Rpc` is already owned by `Boss::Mod::Initiator`, which has a unique pointer to the owned RPC object.

https://github.com/ZmnSCPxj/clboss/blob/d3eb1a190fac7f156495a61ac88a5412b4dbe4ff/Boss/Mod/Initiator.cpp#L63

Basically `Boss::Mod::RpcWrapper` is "borrowing" the object from `Boss::Mod::Initiator`.

We *could* switch to using `std::shared_ptr` but in any case, all modules have the same lifetime anyway, so you at least avoid the refcount overhead and use simple `std::unique_ptr`s which are as cheap as plain pointers, and we already know that this is a lease and the object responsible for cleaning up `Boss::Mod::Rpc` will always be `Boss::Mod::Initiator`..

Reply:
Yeah make sense, this idea to avoid extra overhead is good!

Thanks to explain to me the motivation of your choice!

Reply:
Smart pointers are cool and all, but I have been guilty of overusing `std::shared_ptr` in the past when there was well-defined responsibilities of who owned what objects anyway.  So the use of non-smart pointers should not be too bad.  On the other hand, the entire point of this particular modification is to avoid having a bare pointer to `Boss::Mod::Rpc` kept by multiple modules, which is one reason for the proxy and the wrapper added in this branch.